include/json_struct/json_struct.h

```diff
@@ -4524,6 +4524,88 @@ void populateEnumNames(std::vector<DataRef> &names, const char (&data)[N])
   };                                                                                                                   \
   }
 
+#define JS_ENUM_DECLARE_VALUE_PARSER(name)                                                                             \
+  namespace JS                                                                                                         \
+  {                                                                                                                    \
+  template <>                                                                                                          \
+  struct TypeHandler<name>                                                                                             \
+  {                                                                                                                    \
+    static inline Error to(name &to_type, ParseContext &context)                                                       \
+    {                                                                                                                  \
+      if (std::is_unsigned<name>::value)                                                                               \
```

Comment:
What about using std::underlying_type instead? https://en.cppreference.com/w/cpp/types/underlying_type

Reply:
Yes, std::underlying_type even better. Modern c++ keeps surprising me.

Should I submit another pull request and let you cancel this one? (I'm still learning how to use github)